### PR TITLE
Update in-tree docs to point to the new book

### PR DESCRIPTION
**Pull request by delan**
This patch stubs out all of our in-tree docs to point to [The Servo Book](https://book.servo.org), which is not 100% polished yet but at least [up to date](https://book.servo.org/hacking/older-versions.html). When this lands, I’ll also do the same for the wiki pages:

- [Building](https://github.com/servo/servo/wiki/Building/95a6f67d29c58d9269c55b9128dbf863339c7502)
- [Building for Android](https://github.com/servo/servo/wiki/Building-for-Android/77acd4dd5bfe40aa48d4df967259f89ffadab066)
- [Design](https://github.com/servo/servo/wiki/Design/0941531122361aac8c88d582aa640ec689cdcdd1)
- [Profiling](https://github.com/servo/servo/wiki/Profiling/1963e5c31e3319d7167f29181cbb6229cf7384eb)

---

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they only affect the markdown docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,213 +4,7 @@ Servo is a prototype web browser engine written in the
 [Rust](https://github.com/rust-lang/rust) language. It is currently developed on
 64-bit macOS, 64-bit Linux, 64-bit Windows, and Android.
 
-Servo welcomes contribution from everyone.  See
-[`CONTRIBUTING.md`](CONTRIBUTING.md) and [`HACKING_QUICKSTART.md`](docs/HACKING_QUICKSTART.md)
-for help getting started.
+Servo welcomes contribution from everyone.
 
-Visit the [Servo Project page](https://servo.org/) for news and guides.
+Check out [The Servo Book](https://book.servo.org) to get started, or go to [servo.org](https://servo.org/) for news and guides.
```
Comment (mrobinson) [inline README.md:9]: Would it be possible to keep the basic build instructions in the README? It's pretty typical to have a quick getting started section in a project's README. I'm just talking about everything under the "Build Setup" header. I'm happy to move everything else.

**Review by mrobinson — CHANGES_REQUESTED** [review]
This looks like a great change. Just a couple requests:

1. Can we keep the very basic build instructions in the README?
2. I would just go ahead and remove the now empty markdown files that in this change just have links to the book.

File: README.md (hunk 1 above)
Comment (delan) [reply README.md]: done in 5f83fd4c550740268ec0bd8d01db41b3a149a50f, but i’ve also incorporated some of the updates made in the book, and added an extra step to actually build servoshell

**Comment by delan** [comment]
> 2. I would just go ahead and remove the now empty markdown files that in this change just have links to the book.

I think we should keep them, at least the really conventional ones like CONTRIBUTING.md and (for a while at least) really well-known ones like HACKING_QUICKSTART.md. People are still gonna follow links to those and expect docs, and it’s much more helpful to get a link to the book than to just get a 404 and have to dig through the Git history.

File: tests/wpt/README.md [hunk 2]
```diff
@@ -1,246 +1,6 @@
-This folder contains the web platform tests and the
-code required to integrate them with Servo. 
-To learn how to write tests, go [here](http://web-platform-tests.org/writing-tests/index.html).
+# Web tests
```
Comment (mrobinson) [inline tests/wpt/README.md:1]: This is the proper name of a project so all caps, I think:

```suggestion
# Web Platform Tests
```

File: docs/COMMAND_LINE_ARGS.md [hunk 3]
```diff
@@ -1,32 +1,7 @@
-Command Line Arguments
-========================
-# General
+# Command Line Arguments
```
Comment (mrobinson) [inline docs/COMMAND_LINE_ARGS.md:1]: For consistency: 

```suggestion
# Command line arguments
```

File: README.md [hunk 4]
```diff
@@ -4,75 +4,44 @@ Servo is a prototype web browser engine written in the
 [Rust](https://github.com/rust-lang/rust) language. It is currently developed on
 64-bit macOS, 64-bit Linux, 64-bit Windows, and Android.
 
-Servo welcomes contribution from everyone.  See
-[`CONTRIBUTING.md`](CONTRIBUTING.md) and [`HACKING_QUICKSTART.md`](docs/HACKING_QUICKSTART.md)
-for help getting started.
+Servo welcomes contribution from everyone. Check out [The Servo Book](https://book.servo.org) to get started, or go to [servo.org](https://servo.org/) for news and guides.
 
-Visit the [Servo Project page](https://servo.org/) for news and guides.
+## Quickstart
```
Comment (mrobinson) [inline README.md:9]: Let's call this "Getting started" to avoid jargon:

```suggestion
## Getting started
```

File: docs/ORGANIZATION.md [hunk 5]
```diff
@@ -1,110 +1,6 @@
...
-* <https://github.com/rust-windowing/winit>: cross-platform windowing and input
-* <https://github.com/jrmuizel/raqote>: a pure Rust 2D graphics library
-* <https://github.com/servo/rust-cssparser>: a CSS parser
-* <https://github.com/housleyjk/ws-rs>: a WebSocket protocol implementation
-* <https://github.com/servo/rust-url>: an implementation of the URL specification
-* <https://github.com/servo/webrender>: a GPU renderer
+Moved to the Servo book:
+
+- Servo’s architecture
+  - [Directory structure](https://book.servo.org/architecture/directory-structure.html)
```
Comment (mrobinson) [inline docs/ORGANIZATION.md:6]: Nit: The architecture is different from the directory structure and a separate set of docs, right? Maybe this should just say "directory structure."

**Review by mrobinson — APPROVED** [review]
I'm okay with leaving these around for now, but we should look to remove the files that simply link to the book in the future. I'm not sure how many links their are into the GitHub repository.

File: docs/ORGANIZATION.md (hunk 5 above)
Comment (delan) [reply docs/ORGANIZATION.md:6]: “Servo’s architecture” is the section containing that chapter in the book’s sidebar, it’s just there to help you find the chapter (while being more stable than the chapter number)

File: docs/COMMAND_LINE_ARGS.md (hunk 3 above)
Comment (delan) [reply docs/COMMAND_LINE_ARGS.md]: done in bf938ae761742a4a7ecdf697cbbda17ebc059e9c

Comment (delan) [reply docs/COMMAND_LINE_ARGS.md]: done in bf938ae761742a4a7ecdf697cbbda17ebc059e9c

File: README.md (hunk 4 above)
Comment (delan) [reply README.md]: done in bf938ae761742a4a7ecdf697cbbda17ebc059e9c

File: tests/wpt/README.md (hunk 2 above)
Comment (delan) [reply tests/wpt/README.md]: not all of the tests in tests/wpt are Web Platform Tests, only those in tests/wpt/tests. i’ve updated it to mention both in bf938ae761742a4a7ecdf697cbbda17ebc059e9c

**Comment by delan** [comment]
> I'm okay with leaving these around for now, but we should look to remove the files that simply link to the book in the future. I'm not sure how many links their are into the GitHub repository.

I agree. Using the wiki as an example, I think only the WPT README, HACKING_QUICKSTART.md, and CONTRIBUTING.md are likely to matter in the long run:

```
~/code/servo-wiki % rg -oI 'https://github.com/servo/servo/blob/[^ ]+[.]md' | sort | uniq -c | sort -nr
     14 https://github.com/servo/servo/blob/master/tests/wpt/README.md
      7 https://github.com/servo/servo/blob/master/docs/HACKING_QUICKSTART.md
      7 https://github.com/servo/servo/blob/master/CONTRIBUTING.md
      2 https://github.com/servo/servo/blob/master/README.md
      2 https://github.com/servo/servo/blob/21ea6d21f0f3bcb2e736082e397ef99cf9ecd051/components/script/docs/JS-Servos-only-GC.md
      1 https://github.com/servo/servo/blob/master/docs/hololens.md
      1 https://github.com/servo/servo/blob/master/docs/glossary.md
      1 https://github.com/servo/servo/blob/main/docs/components/webxr.md
```

**Comment by delan** [comment]
Updated wiki pages:

- [Building](https://github.com/servo/servo/wiki/Building/a2b79be0b7f51f73fe8fd3d924919416dc4b5521)
- [Building for Android](https://github.com/servo/servo/wiki/Building-for-Android/59092999d3ef203b824b849f9b26135951c39807)
- [Design](https://github.com/servo/servo/wiki/Design/59f70d8fa14827bcf6d5f113f31e7c9e937e9e02)
- [Profiling](https://github.com/servo/servo/wiki/Profiling/7632dd08544e0b05c9b288eae2cd2d5c7c8fad2e)

File: etc/doc.servo.org/index.html
Comment (sagudev) [inline etc/doc.servo.org/index.html:1]: I think this should remain as it was, doc.servo.org is for: https://doc.servo.org/servo/index.html

Comment (delan) [reply etc/doc.servo.org/index.html:1]: i changed it based on the idea that if someone goes to the root of doc.servo.org, they probably just want general Servo docs, not necessarily the API docs, see also [the root of doc.rust-lang.org](https://doc.rust-lang.org). that said, the wiki and book needs to be updated accordingly, i’ll do that.

Comment (delan) [reply etc/doc.servo.org/index.html:1]: [wiki updated](https://github.com/servo/servo/wiki/Home/_compare/6ad16d61183572526d8092d78ea955d50b333622...97a5c42353b23e2662b3cc900cbcccccd4855fdf), [book updated](https://github.com/servo/book/commit/b1c4b1404470016a10562dc479c45e1104506dbc)

**Pull request merged**